go-controller/pkg/config/config.go

```diff
@@ -831,6 +834,12 @@ var CommonFlags = []cli.Flag{
 		Destination: &cliConfig.Logging.ACLLoggingRateLimit,
 		Value:       20,
 	},
+	&cli.StringFlag{
```

Comment:
can we have some more comments or explanation in in the commit message about why we need to be able to pass the zone name? Is it simply to indicate to ovnkube what zone it should be a part of? If so, then why do we need to pass that via CLI? Why wouldnt the node read what zone it is part of that is assigned to it by cluster manager?

Reply:
Sure.

How I'd see this would work is:

The daemonset yaml file which starts OVN NB DB will set the zone name in the ovn nb db

ovn-nbctl set NB_Global . name=<node_name>
ovn-nbctl set NB_Global . options:name=<node_name> // ovn-northd will copy this to SB_Global options so that ovnkube-node can read this

And the daemonset will pass the --zone option while starting ovnkube-node and ovnkube-ncm

When ovnkube-node comes, it will read the SB DB Zone name and matches with the passed --zone option.
If it matches, it will update the node's annotation - k8s.ovn.org/zone-name with the zone.

When ovnkube-ncm comes up it will read the NB DB zone name and matches with the passed --zone option.
ovnkube-ncm will read each nodes' zone-name annotation to figure out if the node is a  local zone node or a remote zone node.

Please note that cluster manager will not assign the zone name.